cmd/tempo-cli/cmd-query-trace-summary.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math"
+	"sort"
+
+	"github.com/gogo/protobuf/jsonpb"
+
+	"github.com/grafana/tempo/pkg/model/trace"
+	v1resource "github.com/grafana/tempo/pkg/tempopb/resource/v1"
+	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
+	"github.com/grafana/tempo/pkg/util"
+)
+
+type queryTraceSummaryCmd struct {
+	backendOptions
+
+	TraceID  string `arg:"" help:"trace ID to retrieve"`
+	TenantID string `arg:"" help:"tenant ID to search"`
+}
+
+func (cmd *queryTraceSummaryCmd) Run(ctx *globalOptions) error {
+	r, _, c, err := loadBackend(&cmd.backendOptions, ctx)
+	if err != nil {
+		return err
+	}
+
+	id, err := util.HexStringToTraceID(cmd.TraceID)
+	if err != nil {
+		return err
+	}
+
+	results, err := queryBucket(context.Background(), r, c, cmd.TenantID, id)
+	if err != nil {
+		return err
+	}
+
+	var (
+		combiner   = trace.NewCombiner()
+		marshaller = new(jsonpb.Marshaler)
+		jsonBytes  = bytes.Buffer{}
+	)
+
+	fmt.Println()
+	for i, result := range results {
+		combiner.ConsumeWithFinal(result.trace, i == len(results)-1)
+	}
+
+	combinedTrace, _ := combiner.Result()
+
+	var rootSpan *v1.Span
+	var rootSpanResource *v1resource.Resource
+
+	size := 0
+	spanCount := 0
+
+	firstStartTime := uint64(math.MaxUint64)
+	lastEndTime := uint64(0)
+
+	rootServiceName := ""
+	serviceNameMap := make(map[string]int)
+
+	for _, b := range combinedTrace.Batches {
+		size += b.Size()
+		for _, attr := range b.Resource.Attributes {
+			if "service.name" == attr.Key {
+				serviceNameMap[attr.Value.GetStringValue()]++
+			}
+		}
+		for _, scope := range b.ScopeSpans {
+			spanCount += len(scope.Spans)
+			for _, span := range scope.Spans {
+				if span.StartTimeUnixNano < firstStartTime {
+					firstStartTime = span.StartTimeUnixNano
+				}
+				if span.EndTimeUnixNano > lastEndTime {
+					lastEndTime = span.EndTimeUnixNano
+				}
+				if len(span.ParentSpanId) == 0 {
+					rootSpan = span
+					rootSpanResource = b.Resource
+				}
+			}
+		}
+	}
+
+	for _, attr := range rootSpanResource.Attributes {
+		if "service.name" == attr.Key {
+			rootServiceName = attr.Value.GetStringValue()
+		}
+	}
+
+	// get top 5 most frequent service names
+	topFiveSortedPL := sortServiceNames(serviceNameMap)
+	topFiveServiceName := make([]string, 5)
+	length := len(topFiveSortedPL)
```

Comment:
what if length is less than 5? i think this panics

Reply:
since we are iterating through the actual list of service names using its length (or if it's longer than 5 use 5), even if it's less than 3, we wouldn't get to `out of range`.  It just prints out a weird looking array with extra spaces

`[a b c      ]`